incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -15,6 +15,201 @@ import (
 	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
 )
 
+var _ = Describe("Exception", func() {
+	ctx := context.Background()
+
+	BeforeEach(func() {
+		// We want to ensure we're working with a clean slate, in case a previous tests objects still exist
+		cleanupObjects(ctx)
+	})
+
+	AfterEach(func() {
+		// Change current config back to the default value.
+		resetHNCConfigToDefault(ctx)
+		cleanupObjects(ctx)
+	})
+
+	Context("Add exception annotations ", func() {
+		const (
+			p  = "parent"
+			c1 = "child1"
+			c2 = "child2"
+			c3 = "child3"
+		)
+		tests := []struct {
+			name         string
+			selector     string
+			treeSelector string
+			want         []string
+			notWant      []string
+		}{
+			{name: "simply using select",
+				selector:     "!" + c1 + api.LabelTreeDepthSuffix,
+				treeSelector: "",
+				want:         []string{c2, c3},
+				notWant:      []string{c1},
```

Comment:
Maybe calculate `notWant` from `want`?

Reply:
I did a brief search and seems like the only way is to iterate through the whole set (all three nses) for each element in the `want`. That looks a bit overkill for me (?) So I thought since there are only three elements we might just keep both sets

Reply:
```
for _, ns := range []string{c1, c2, c3} {
  if isIn(ns, tc.want) {
    ... want case ...
  } else {
    .. not-want case ...
  }
}
```

?

Reply:
But you can do that in a future PR, it doesn't have to be in this one